app/src/main/java/br/digitalhouse/marveltime/LoginActivity.java

```diff
@@ -0,0 +1,160 @@
+package br.digitalhouse.marveltime;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.text.Spannable;
+import android.text.method.LinkMovementMethod;
+import android.text.style.ClickableSpan;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.android.material.textfield.TextInputLayout;
+
+public class LoginActivity extends AppCompatActivity {
+
+    private TextInputLayout loginUsuario;
+    private TextInputLayout loginSenha;
+    private FloatingActionButton bntLogin;
+    private TextView loginRegistro;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        //INGHRIDY - 03/02/20 - SET01- Criação das Branchs GITHUB
+        Log.i("Teste", "Teste");
+        Log.i("Teste", "Teste - branch epic");
+
+        initViews();
+
+        bntLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                String usuario = loginUsuario.getEditText().getText().toString();
+                String senha = loginSenha.getEditText().getText().toString();
+
+                if (usuarioVazio(usuario) || senhaVazia(senha))
+                    notificacao();
+                else if (!usuarioValido(usuario) || !senhaValida(senha))
+                    notificacaoInformacaoIncorreta();
+                else
+                    notificacaoParaProximaTela();
+            }
+        });
+
+        linkCadastroUsuario();
+
+    }
+
+
+    //INGHRIDY SANTOS 04/02/20 - BDEV04 - TEXTVIEW COM LINK
+    private void linkCadastroUsuario() {
+        String msgRegistroUsuario = "Ainda não é membro? [Registre-se aqui]";
+        int i1 = msgRegistroUsuario.indexOf("[");
```

Comment:
A declaração das variáveis int  i1 e int i2 ficaram genéricas, poderiam ser declaradas de uma forma mais fácil de ser entendidas.

Reply:
Acho coerente manter i1 e i2, pois i significa index. Ou seja, estou falando de entre a posição index 1 e index 2, existem um deste que deve ser um link.

A função do i neste contexto é a mesma do i dentro de um for.